gateway/api/user/user.go

```diff
@@ -581,6 +581,177 @@ func ListAllGroups(c *gin.Context) {
+//	@Router			/users/groups/{name} [put]
+func UpdateGroup(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+	oldName := c.Param("name")
+
+	var req openapi.UserGroup
+
+	if err := c.ShouldBindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": err.Error()})
+		return
+	}
+
+	// Check if group exists
+	userGroups, err := models.GetUserGroupsByOrgID(ctx.OrgID)
+	if err != nil {
+		log.Errorf("failed getting org groups, err=%v", err)
+		sentry.CaptureException(err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "failed getting org groups"})
+		return
+	}
+
+	groupExists := false
+	newNameExists := false
+	for _, ug := range userGroups {
+		if ug.Name == oldName {
+			groupExists = true
+		}
+		if ug.Name == req.Name && oldName != req.Name {
+			newNameExists = true
+		}
+	}
+
+	if !groupExists {
+		c.JSON(http.StatusNotFound, gin.H{"message": fmt.Sprintf("group %s not found", oldName)})
+		return
+	}
+
+	if newNameExists {
+		c.JSON(http.StatusConflict, gin.H{"message": fmt.Sprintf("group %s already exists", req.Name)})
+		return
+	}
+
+	// Update all instances of this group
+	if err := models.UpdateUserGroupName(ctx.OrgID, oldName, req.Name); err != nil {
+		log.Errorf("failed updating group, err=%v", err)
+		sentry.CaptureException(err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "failed updating group"})
+		return
+	}
+
+	c.JSON(http.StatusOK, req)
+}
+
+// DeleteGroup
+//
+//	@Summary		Delete User Group
+//	@Description	Delete a user group
+//	@Tags			User Management
+//	@Produce		json
+//	@Param			name	path	string	true	"The name of the group to delete"
+//	@Success		204		"No Content"
+//	@Failure		404		{object}	openapi.HTTPError	"{"message": "group not found"}"
+//	@Failure		422		{object}	openapi.HTTPError	"{"message": "cannot delete built-in group"}"
+//	@Failure		500		{object}	openapi.HTTPError	"{"message": "server error"}"
+//	@Router			/users/groups/{name} [delete]
+func DeleteGroup(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+	name := c.Param("name")
+
+	// Prevent deletion of built-in groups
+	if name == types.GroupAdmin || name == types.GroupAuditor {
+		c.JSON(http.StatusUnprocessableEntity, gin.H{"message": fmt.Sprintf("cannot delete built-in group %s", name)})
+		return
+	}
+
+	// Check if group exists
+	userGroups, err := models.GetUserGroupsByOrgID(ctx.OrgID)
+	if err != nil {
+		log.Errorf("failed getting org groups, err=%v", err)
+		sentry.CaptureException(err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "failed getting org groups"})
+		return
+	}
+
+	groupExists := false
+	for _, ug := range userGroups {
+		if ug.Name == name {
+			groupExists = true
+			break
+		}
+	}
+
+	if !groupExists {
+		c.JSON(http.StatusNotFound, gin.H{"message": fmt.Sprintf("group %s not found", name)})
+		return
+	}
+
+	// Delete all instances of this group
+	if err := models.DeleteUserGroup(ctx.OrgID, name); err != nil {
```

Comment:
This function could just return models.ErrNotFound in case it didn't returned any rows in the delete operation